cachi2/core/models/input.py

```diff
@@ -172,11 +117,62 @@ def _requirements_file_path_is_relative(cls, paths: list[Path]) -> list[Path]:
         return paths
 
 
+class _ExtraOptions(pydantic.BaseModel, extra="forbid"):
+    """Global package manager extra options model.
+
+    This model takes care of carrying and parsing various kind of extra options that need to be
+    passed through to CLI commands/services we interact with underneath to tweak their behaviour
+    rather than our own. Each option set is namespaced by the corresponding tool/service it is
+    related to.
+
+    TODO: Enable this globally for all pkg managers not just the RpmPackageInput model.
+    """
+
+    dnf: Optional[Dict[Union[Literal["main"], str], Dict[str, Any]]] = None
+
+    @pydantic.model_validator(mode="before")
+    def _validate_dnf_options(cls, data: Any, info: pydantic.ValidationInfo) -> Any:
```

Comment:
If we are to reuse this with other PMs then we should better define `_ExtraOptions` as a union type of individual PMs options. This is not a blocker and could be done when such need arises.

Reply:
> If we are to reuse this with other PMs then we should better define _ExtraOptions as a union type of individual PMs options. This is not a blocker and could be done when such need arises.

I don't follow - that's how it was the previously declared and it was a PITA to integrate with it. What would the Union bring over an explicitly enumerated list of keywords (e.g. `dnf`, `ssl`, `foo`) each defined as `Optional` further specified model?